src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -4290,7 +4290,7 @@ internal static unsafe Vector<T> SquareRoot(Vector<T> value)
                     SByte* dataPtr = stackalloc SByte[Count];
                     for (int g = 0; g < Count; g++)
                     {
-                        dataPtr[g] = (SByte)Math.Sqrt((SByte)(object)value[g]);
+                        dataPtr[g] = unchecked((SByte)Math.Sqrt((SByte)(object)value[g]));
```

Comment:
I'm fine with having it in debug builds; we can continue the discussion if/when we think about turning it on for release builds.